external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -147,6 +147,21 @@ private[sql] class AvroDeserializer(
           s"Avro logical type $other cannot be converted to SQL type ${TimestampType.sql}.")
       }
 
+      case (LONG, TimestampNTZType) => avroType.getLogicalType match {
+        // For backward compatibility, if the Avro type is Long and it is not logical type
+        // (the `null` case), the value is processed as timestamp without time zone type
+        // with millisecond precision.
+        case null | _: LocalTimestampMillis => (updater, ordinal, value) =>
+          val millis = value.asInstanceOf[Long]
+          val micros = DateTimeUtils.millisToMicros(millis)
+          updater.setLong(ordinal, timestampRebaseFunc(micros))
```

Comment:
It seems that we don't need to rebase here. 
cc @MaxGekk

Reply:
ping @MaxGekk too.

Reply:
I think we need to rebase. Let's look at it. We save/load some offset for a local timestamp, the same local timestamp have different UTC offsets in Proleptic Gregorian and Julian calendar because years have different duration in the calendars.

Reply:
Hmmm, the TimestampNTZ type is independent of time zones offsets. On read/write as long type, it is treated as in  UTC time zone. So it seems we don't need the rebase.
We need to figure this out so that we can follow it in other data source support works.
cc @cloud-fan as well

Reply:
> the TimestampNTZ type is independent of time zones offsets ...

I mean the offset from the epoch.

The rebasing is not needed only if we declare that microseconds since the epoch (in UTC) saved to Avro represent a local timestamp in Proleptic Gregorian calendar.

Reply:
Let's keep it simple and get rid of rebasing. We can add documents saying that TimestampNTZ represents a local timestamp in Proleptic Gregorian calendar later.

Reply:
It goes without saying that a modern system always uses Proleptic Gregorian calendar. The rebase is only used to keep backward compatibility with legacy Spark versions. This is a new data type and there is no backward compatibility issues.